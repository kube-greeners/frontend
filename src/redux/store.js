import { configureStore } from '@reduxjs/toolkit'
import dashboardReducer from './dashboardSlice'
import { apiSlice } from './apiSlice'

export default configureStore({
  reducer: {
    dashboard: dashboardReducer,
    [apiSlice.reducerPath]:apiSlice.reducer
  },
  middleware: getDefaultMiddleware =>
  getDefaultMiddleware().concat(apiSlice.middleware)
})