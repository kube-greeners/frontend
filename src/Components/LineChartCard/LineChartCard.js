import React from "react";
import LineChart from "../LineChart/LineChart";
import LoadingSpinner from "../LoadingSpinner/LoadingSpinner";
import { setCurrentlySelectedResource } from "../../redux/dashboardSlice";
import { useDispatch, useSelector } from "react-redux";
import { Selector } from "./../Selectors/Selectors";
import { Card } from "antd";
​
function LineChartCard({ dataFetching, currentlyShowing }) {
  const { resources } = useSelector((state) => state.dashboard.selects);
​
  const dispatch = useDispatch();
​
  const resourceSelected = (rs) => {
    dispatch(setCurrentlySelectedResource(rs));
  };
​
  const labelStyle = {
    marginBottom: ".5rem",
    display: "block",
    color: "#666666",
  };
​
  return (
    <Card
      style={{ gridArea: "lc" }}
      title={
        <Selector
          title={false}
          data={resources.data}
          name="Resource"
          defaultVal={resources.currentlySelected}
          labelStyle={labelStyle}
          style={{ display: "block" }}
          onChange={resourceSelected}
        />
      }
    >
      {!dataFetching.isFetching && dataFetching.isSuccess ? (
        <LineChart data={dataFetching.data.historicalData} />
      ) : (
        <div
          style={{
            height: 500,
            display: "flex",
            justifyContent: "center",
            alignItems: "center",
          }}
        >
          <LoadingSpinner />
        </div>
      )}
    </Card>
  );
}
​
export default LineChartCard;