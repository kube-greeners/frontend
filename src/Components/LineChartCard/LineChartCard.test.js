import LineChartCard from "./LineChartCard";
import renderer from "react-test-renderer";
import store from "../../redux/store";
import { Provider } from "react-redux";

test("matches snapshot", () => {
  const currentlyShowing = "production";
  const dataFetching = {
    status: "pending",
    endpointName: "getCO2Emission",
    requestId: "Rxqq7eCVdUEv-vzxZMo1q",
    originalArgs: {
      namespace: "production",
      interval: "5d",
      step: "1h",
    },
    startedTimeStamp: 1639324236170,
    isUninitialized: false,
    isLoading: true,
    isSuccess: false,
    isError: false,
    isFetching: true,
  };
  const tree = renderer
    .create(
      <Provider store={store}>
        <LineChartCard currentlyShowing={currentlyShowing}
        dataFetching={dataFetching} />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
