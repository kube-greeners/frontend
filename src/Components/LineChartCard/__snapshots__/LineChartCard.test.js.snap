// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="ant-card ant-card-bordered"
  style={
    Object {
      "gridArea": "lc",
    }
  }
>
  <div
    className="ant-card-head"
    style={Object {}}
  >
    <div
      className="ant-card-head-wrapper"
    >
      <div
        className="ant-card-head-title"
      >
        production
      </div>
    </div>
  </div>
  <div
    className="ant-card-body"
    style={Object {}}
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "height": 500,
          "justifyContent": "center",
        }
      }
    >
      <div
        className="lds"
      >
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
      </div>
    </div>
  </div>
</div>
`;
